packages/next/src/build/webpack-config.ts

```diff
@@ -96,8 +97,12 @@ const babelIncludeRegexes: RegExp[] = [
 
 const reactPackagesRegex = /^(react|react-dom|react-server-dom-webpack)($|\/)/
 
-const asyncStoragesRegex =
-  /next[\\/]dist[\\/]client[\\/]components[\\/](static-generation-async-storage|action-async-storage|request-async-storage)/
+const clientLayerRegex =
+  /next[\\/]dist[\\/](client|server)[\\/].*?(?:\.|[\\/])clientlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
+const serverLayerRegex =
+  /next[\\/]dist[\\/](client|server)[\\/].*?(?:\.|[\\/])serverlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
+const sharedLayerRegex =
+  /next[\\/]dist[\\/](client|server)[\\/].*?(?:\.|[\\/])sharedlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
```

Comment:
```suggestion
const clientLayerRegex =
  /next[\\/]dist[\\/](esm[\\/])?(client|server)[\\/].*?(?:\.|[\\/])clientlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
const serverLayerRegex =
  /next[\\/]dist[\\/](esm[\\/])?(client|server)[\\/].*?(?:\.|[\\/])serverlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
const sharedLayerRegex =
  /next[\\/]dist[\\/](esm[\\/])?(client|server)[\\/].*?(?:\.|[\\/])sharedlayer(?:$|(?:\.(tsx|ts|js|cjs|mjs|jsx))|\\|\/)/
```

As Next.js has 2 builds (CJS and ESM).

Reply:
shoudl be good now